### ISSUE TITLE
E731 should not apply when assigning to a field

### ISSUE THREAD

**Issue by mypalmike** [PyCQA/pycodestyle#311]
For example, this code is commonly used for setting the http method in a urllib2 request:

request.get_method = lambda: 'DELETE'

The library is expecting a function, and using a lambda here is not against the PEP8 intent as far as I can tell, which is binding a lambda to a name. This could be done with a def, but the code here is simpler (inasmuch as people find lambda to be simple).

**Comment by kartynnik** [comment]
What about the legitimate case
```python
return lambda: whatever
```
?
This is useful in various forms of currying and binding like
```python
def _bind(self, func):
     '''Takes an unbound method and binds it to :py:var:`self`.'''
     return lambda *args, **kwargs: func(self, *args, **kwargs)
```

**Comment by sigmavirus24** [comment]
That example can be fixed by using `functools.partial`

**Comment by jayvdb** [comment]
While functools.partial might be a workaround in some cases, that isnt the issue.  Using return does not bind the lambda to an identifier.

**Comment by kartynnik** [comment]
@jayvdb, exactly. @sigmavirus24, it's a workaround for the pep8 behavior, not a fix, because there's nothing violating PEP8 there (we're not binding the lambda to an identifier and we can't rephrase that with a def of the same identifier - we have to think over an intermediate name). Actually, this workaround is not universal either: one might want to return not a partial application of a function, but a completely fresh function that is a closure over the locals; using a nested def looks times uglier and confuses structure folding in some IDEs.

**Comment by kartynnik** [comment]
I would rephrase the point of PEP8 like this: lambdas are for unnamed functions and it's pointless to use them when you actually *need* and *provide* a name via a *direct assignment* to an identifier (which is really more readable when phrased as a def).
